eth/stagedsync/stage_execute.go

```diff
@@ -197,7 +197,14 @@ func unwindExec3(u *UnwindState, s *StageState, txc wrap.TxContainer, ctx contex
 			}
 		}
 	}
-	if err := rs.Unwind(ctx, txc.Tx, u.UnwindPoint, txNum, accumulator, changeset); err != nil {
+
+	// if the changeset is nil here, no unwinding took place due to the loop above being empty
+	if changeset == nil { // nothing to unwind, exit function prematurely
+		logger.Warn("unwindExec3: nothing to unwind")
```

Comment:
i think we must return error in this case. because we have to unwind, but can't - because not enough changesets in db.